Calculator.py

```diff
@@ -0,0 +1,103 @@
+                                                "%": lambda num1, num2: num1 % num2}
+        self.dict_of_operations_prioritize_3 = {"^": lambda num1, num2: num1 ** num2}
+        self.dict_of_operations_prioritize_2 = {"*": lambda num1, num2: num1 * num2,
+                                                "/": lambda num1, num2: num1 / num2}
+        self.dict_of_operations_prioritize_1 = {"+": lambda num1, num2: num1 + num2,
+                                                "-": lambda num1, num2: num1 - num2}
+        self.list_of_prioritize = [self.dict_of_operations_prioritize_5, self.dict_of_operations_prioritize_4,
+                                   self.dict_of_operations_prioritize_3, self.dict_of_operations_prioritize_2,
+                                   self.dict_of_operations_prioritize_1]
+
+    def change_list(self, result, index: int):
+        """The function changing list values after calculate"""
+        self.list_of_exercise[index] = result
+        self.list_of_exercise.pop(index + 1)
+        self.list_of_exercise.pop(index - 1)
+
+    def exercise_handler(self):
+        """ This function handel in use of [ ] in the exercise and using calculate method"""
+        while "[" in self.list_of_exercise:
+            temp_list = []
+            opening = len(self.list_of_exercise) - self.list_of_exercise[::-1].index("[") - 1
+            closing = self.list_of_exercise.index("]")
+            index = closing
+            while index >= opening:
+                temp_list.insert(0, self.list_of_exercise.pop(index))
+                index -= 1
+            temp_list.pop(0)
+            temp_list.pop(len(temp_list) - 1)
+            temp_instance_of_calc = Calculator(temp_list)
+            result = temp_instance_of_calc.calculate()
+            self.list_of_exercise.insert(opening, result)
+        else:
+            self.calculate()
+
+    def calculate(self):
+        """The function calculating exercise according specific math order"""
+        for dictionary in self.list_of_prioritize:
+            i = 0
+            while i < len(self.list_of_exercise):
+                sign = self.list_of_exercise[i]
+                if sign == "!":
+                    result = dictionary[sign](int(self.list_of_exercise[self.list_of_exercise.index(sign) - 1]))
+                    self.change_list(result, self.list_of_exercise.index(sign))
+                    i -= 1
+                elif sign in dictionary.keys():
+                    result = dictionary[sign](int(self.list_of_exercise[self.list_of_exercise.index(sign) - 1]),
+                                              int(self.list_of_exercise[self.list_of_exercise.index(sign) + 1]))
+                    self.change_list(result, self.list_of_exercise.index(sign))
+                    i -= 2
+                i += 1
+        return self.list_of_exercise[0]
+
+
+def main():
```

Comment:
The main is pretty massive.
The main must be between 5-7 lines, If it's more than this you should split it to another function.